packages/@aws-cdk/aws-cloudwatch/lib/metric-util.ts

```diff
@@ -0,0 +1,308 @@
+import { Duration } from "@aws-cdk/core";
+import { MathExpression } from "./metric";
+import { IMetric, MetricConfig, MetricExpressionConfig, MetricStatConfig } from "./metric-types";
+
+/**
+ * Return the JSON structure which represents these metrics in a graph.
+ *
+ * Depending on the metric type (stat or expression), one `Metric` object
+ * can render to multiple time series.
+ *
+ * - Top-level metrics will be rendered visibly, additionally added metrics will
+ *   be rendered invisibly.
+ * - IDs used in math expressions need to be either globally unique, or refer to the same
+ *   metric object.
+ *
+ * This will be called by GraphWidget, no need for clients to call this.
+ */
+export function allMetricsGraphJson(left: IMetric[], right: IMetric[]): any[][] {
+  // Add metrics to a set which will automatically expand them recursively,
+  // making sure to retain conflicting the visible one on conflicting metrics objects.
+  const mset = new MetricSet<string>();
+  mset.addPrimary('left', ...left);
```

Comment:
Could we use `TopLevel` instead of `Primary` everywhere, I have a feeling that `primary` could be interpreted to raw metric (a regular one) rather than a top-level one.